label_studio_converter/converter.py

```diff
@@ -223,52 +224,65 @@ def iter_from_dir(self, input_dir):
+
+    def annotation_result_from_task(self, task):
+        has_annotations = 'completions' in task or 'annotations' in task
+        if not has_annotations:
+            raise KeyError('Each task dict item should contain "annotations" or "completions" [deprecated],'
                            'where value is list of dicts')
-        result = []
-        if has_annotations:
-            # get last not skipped completion and make result from it
-            annotations = d.get('annotations')
-            if annotations is None:
-                annotations = d['completions']
-            tmp = list(filter(lambda x: not (x.get('skipped', False) or x.get('was_cancelled', False)), annotations))
-            if len(tmp) > 0:
-                # TODO: only one annotation per task is supported for non full JSON formats
-                result = sorted(tmp, key=lambda x: x.get('created_at', 0), reverse=True)[0]['result']
-            else:
-                return None
-
-        elif 'result' in d:
-            result = d['result']
-        inputs = d['data']
-        outputs = defaultdict(list)
-        for r in result:
-            if 'from_name' in r and r['from_name'] in self._output_tags:
-                v = deepcopy(r['value'])
-                v['type'] = self._schema[r['from_name']]['type']
-                if 'original_width' in r:
-                    v['original_width'] = r['original_width']
-                if 'original_height' in r:
-                    v['original_height'] = r['original_height']
-                outputs[r['from_name']].append(v)
-        return {
-            'id': d['id'],
-            'input': inputs,
-            'output': outputs
-        }
+
+        # get last not skipped completion and make result from it
+        annotations = task['annotations'] if 'annotations' in task else task['completions']
+        
+        # skip cancelled annotations
+        cancelled = lambda x: not (x.get('skipped', False) or x.get('was_cancelled', False))
+        annotations = list(filter(cancelled, annotations))
+        if not annotations:
+            return None
+        
+        # sort by creation time
+        annotations = sorted(annotations, key=lambda x: x.get('created_at', 0), reverse=True)
+
+        for annotation in annotations:
+            inputs = task['data']
+            result = annotation['result']
+            outputs = defaultdict(list)
+
+            # get results only as output
+            for r in result:
+                if 'from_name' in r and r['from_name'] in self._output_tags:
+                    v = deepcopy(r['value'])
+                    v['type'] = self._schema[r['from_name']]['type']
+                    if 'original_width' in r:
+                        v['original_width'] = r['original_width']
+                    if 'original_height' in r:
+                        v['original_height'] = r['original_height']
+                    outputs[r['from_name']].append(v)
+
+            yield {
```

Comment:
Why we need this conversion? Maybe it would be easier when each converter gets data with common label studio tasks format?